controllers/front/CartController.php

```diff
@@ -135,19 +135,13 @@ public function displayAjaxUpdate()
         $productQuantity = $updatedProduct['quantity'] ?? 0;
 
         if (!$this->errors) {
-            $presentedCart = $this->cart_presenter->present($this->context->cart, true);
-
-            // filter product output
-            $presentedCart['products'] = $this->get('prestashop.core.filter.front_end_object.product_collection')
-                ->filter($presentedCart['products']);
-
```

Comment:
I had to remove this, because setting a value on the lazy array makes a fatal error

Also since this is a lazy array, data is fetched on demand, no need to preset the data.

Reply:
(answered, exceptionally we'll add a setter for this filtered data)

Reply:
@matthieu-rolland If you make it extend AbstractLazyArray, you can just do

```
$presentedCart->offsetUnset('key', true);
$presentedCart->__set('key', $dataToAssign);
```

But I am not really sure what this piece of code should do.

Reply:
yes we just saw that...  but today modules can update this "array" directly, we want to avoid a breaking change on this, especially because this is passed to a hook

right now it works only if there was no previously changed data, otherwise it triggers a fatal error, this is a BC that could be annoying

(the filter's purpose is to send back only required data I guess)

Reply:
(this will be done in the next Pull Request)

Reply:
I want to make another class that extends `AbstractLazyArray` to modify the `__set` method (wether we unset data or pass the `$force` paramater at true in it does the same thing I think ?)

So that we can allow overriding values only on the presenters we want (by default you can't override and it's a wanted behavior I think)

Reply:
If it's a class using the LazyArray feature, you can't simply use this ?

`$presentedCart->offsetSet('products', $this->get('prestashop.core.filter.front_end_object.product_collection')
                ->filter($presentedCart['products']))`

Setting the third params as `true` will replace the possible existing methods `getProducts()`on this LazyArray by it's simple value.

Also, there is a method called `presentProducts` inside the `CartPresenter`; it's params is an array of products rows. Maybe you can call it with the filtered products ?
(It's only a suggestion, I don't check really more about it, I hope it's not a false way to go :))

Reply:
hello !

well it's a matter of bc break, the cart presenter used to be an array, so people could update its value simply by setting the values on the array.

Now that it's a lazy array, it's an object that behaves like an array, except you can't rewrite already set values, which is a BC break, especially when this object is passed to the CartPresenter's hook

so I'd want it to work just like before for devs, without having to call `offsetSet` directly